src/main/kotlin/no/nav/syfo/util/SykmeldingUtil.kt

```diff
@@ -23,3 +24,43 @@ fun harPeriodeSomMatcherAvstandTilnaermestePeriode(sykmelding: Sykmelding, aktiv
             .map { periode -> antallDager(periode.fom, aktivitetskravdato) }
             .any { avstand -> avstand == avstandTilnaermestePeriode }
 }
+
+fun hentSenesteTOM(sykmeldingDokument: Sykmelding): LocalDate {
+    val nyestePeriodeFoerst: List<Periode> = nyestePeriodeFoerst(sykmeldingDokument.perioder)
+    return hentNyestePeriode(nyestePeriodeFoerst).tom
+}
+
+fun hentNyestePeriode(perioder: List<Periode>): Periode {
+    return nyestePeriodeFoerst(perioder)
+            .stream()
+            .findFirst().get()
+}
+
+fun nyestePeriodeFoerst(perioder: List<Periode>): List<Periode> {
+    return perioder.stream()
+            .sorted(nyestePeriodeFoerst())
+            .collect(Collectors.toList())
+}
+
+private fun nyestePeriodeFoerst(): Comparator<Periode>? {
+    return Comparator { p1: Periode, p2: Periode ->
+        val i = p2.fom.compareTo(p1.fom)
+        if (i == 0) p2.tom.compareTo(p1.tom) else i
+    }
+}
+
+fun hentTidligsteFOM(sykmeldingDokument: Sykmelding): LocalDate {
+    return hentTidligsteFOMFraPerioder(sykmeldingDokument.perioder)
+}
+
+fun hentTidligsteFOMFraPerioder(perioder: List<Periode>): LocalDate {
+    return eldstePeriodeFOM(perioder)
+}
+
+fun eldstePeriodeFOM(perioder: List<Periode>): LocalDate {
+    return perioder.stream().sorted(eldstePeriodeFoerst()).findFirst().orElseThrow { RuntimeException("Mangler periode!") }.fom
```

Comment:
Er det nødvendig å kast exception her eller kan man gjøre sånn?
```suggestion
   return perioder.sortedWith(eldstePeriodeFoerst()).first().fom
```